core/trino-main/src/main/java/io/trino/metadata/TableMetadata.java

```diff
@@ -22,35 +22,19 @@
 
 import static java.util.Objects.requireNonNull;
 
-public class TableMetadata
+public record TableMetadata(CatalogName catalogName, ConnectorTableMetadata metadata)
 {
-    private final CatalogName catalogName;
-    private final ConnectorTableMetadata metadata;
-
-    public TableMetadata(CatalogName catalogName, ConnectorTableMetadata metadata)
+    public TableMetadata
     {
         requireNonNull(catalogName, "catalogName is null");
         requireNonNull(metadata, "metadata is null");
-
-        this.catalogName = catalogName;
-        this.metadata = metadata;
     }
 
     public QualifiedObjectName getQualifiedName()
     {
         return new QualifiedObjectName(catalogName.toString(), metadata.getTable().getSchemaName(), metadata.getTable().getTableName());
     }
 
-    public CatalogName getCatalogName()
-    {
-        return catalogName;
-    }
-
-    public ConnectorTableMetadata getMetadata()
```

Comment:
when converting TableMetadata to record, other getter-like methods should IMO be converted to, otherwise the class API becomes inconsistent: getQualifiedName, getTable, getColumns


after applying, move to separate PR pelase